internal/server/api_snapshots.go

```diff
@@ -79,3 +84,86 @@ func convertSnapshotManifest(m *snapshot.Manifest) *serverapi.Snapshot {
 
 	return e
 }
+
+func (s *Server) handleSnapshotDelete(ctx context.Context, r *http.Request, body []byte) (interface{}, *apiError) {
+	oidstr := mux.Vars(r)["snapshotID"]
+	rep := s.rep
+
+	oid, err := object.ParseID(oidstr)
```

Comment:
this is actually not correct - you're parsing the object ID then casting to manifest.ID, we should always treat those as manifest IDs (which don't require any parsing).

Reply:
BTW. I see what you're trying to do - have fuzzy notion of "object ID", but this is an API and I think we should be precise and always require manifest IDs - the only caller will be the UI and they will already have the manifest ID ready.